File: makeMaxNumber/src/main/kotlin/makeMaxNumber.java
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        StringBuffer sb = new StringBuffer();
        //sb.append(number);
        int index = 0;
        for(int i = 0; i < number.length()-k;i++)
        {
            int max = Integer.MIN_VALUE;
            for(int j = index ; j <= i+k; j++)
            {
                if(max < number.charAt(j)-'0')
                {
                    max = number.charAt(j)-'0';
                    index = j+1;
                }
            }
            sb.append(max);
        }
        return sb.toString();
    }
}

File: makeMaxNumber/src/main/kotlin/makeMaxNumber.kt
class Solution {
    fun solution(number: String, k: Int): String {
        var answer = ""
        val sb = StringBuilder()
        var index = 0
        for (i in 0 until number.length - k) {
            var max = Int.MIN_VALUE
            for (j in index..i + k) {
                if (max < number[j].toInt() - '0'.toInt()) {
                    max = number[j].toInt() - '0'.toInt()
                    index = j + 1
                }
            }
            sb.append(max)
        }
        answer = sb.toString()
        return answer
    }
}